src/ln/channelmanager.rs

```diff
@@ -3441,6 +3441,82 @@ mod tests {
 		assert_eq!(nodes[1].node.list_channels().len(), 0);
 	}
 
+	#[test]
+	fn channel_monitor_claim_revoked_commitment_tx() {
+
+		let nodes = create_network(4);
+
+		// Create some new channel:
+		let chan_1 = create_announced_chan_between_nodes(&nodes, 0, 1);
+
+		// Rebalance the network to generate htlc in the two directions
+		send_payment(&nodes[0], &vec!(&nodes[1])[..], 8000000);
+		// node[0] is gonna to revoke an old state thus node[1] should be able to claim both offered/received HTLC outputs on top of commitment tx
+		let payment_preimage_1 = route_payment(&nodes[0], &vec!(&nodes[1])[..], 3000000).0;
+		let _payment_preimage_2 = route_payment(&nodes[1], &vec!(&nodes[0])[..], 3000000).0;
+
+		// Get the will-be-revoked local txn from node[0]
+		let revoked_local_txn = nodes[0].node.channel_state.lock().unwrap().by_id.iter().next().unwrap().1.last_local_commitment_txn.clone();
```

Comment:
Note that chan_1.3 is the channel_id, so you can do a get(chan_1.3).unwrap() here for a bit more readability.

Reply:
Thanks, updated